core/src/main/java/org/frankframework/http/authentication/SamlAssertionOauth.java

```diff
@@ -0,0 +1,257 @@
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.xml.security.algorithms.MessageDigestAlgorithm;
+import org.apache.xml.security.signature.XMLSignature;
+import org.apache.xml.security.transforms.Transforms;
+
+import org.frankframework.util.XmlUtils;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.encryption.EncryptionException;
+import org.frankframework.encryption.PkiUtil;
+import org.frankframework.http.AbstractHttpSession;
+
+public class SamlAssertionOauth extends AbstractOauthAuthenticator {
+
+	private static final String SAML2_BEARER_GRANT_TYPE = "urn:ietf:params:oauth:grant-type:saml2-bearer";
+	private static final String namespaceURI = "urn:oasis:names:tc:SAML:2.0:assertion";
+
+	private PrivateKey privateKey;
+	private X509Certificate certificate;
+
+	static {
+		org.apache.xml.security.Init.init();
+	}
+
+	public SamlAssertionOauth(AbstractHttpSession session) throws HttpAuthenticationException {
+		super(session);
+	}
+
+	@Override
+	public void configure() throws ConfigurationException {
+		if (session.getClientId() == null) {
+			throw new ConfigurationException("clientId is required");
+		}
+
+		if (session.getClientSecret() == null) {
+			throw new ConfigurationException("clientSecret is required");
+		}
+
+		try {
+			privateKey = PkiUtil.getPrivateKey(session, "Creation of SAML assertion");
+
+			Certificate certificate = PkiUtil.getCertificate(session, "Creation of SAML assertion");
+
+			if (certificate instanceof X509Certificate x509certificate) {
+				this.certificate = x509certificate;
+			} else {
+				throw new ConfigurationException("Certificate must be a X.509 certificate");
+			}
+		} catch (EncryptionException e) {
+			throw new ConfigurationException(e);
+		}
+	}
+
+	private String createAssertion() throws Exception {
+		// Generate SAML Assertion
+		Document samlAssertion = generateSAMLAssertion();
+		String docAsXml = documentToString(samlAssertion);
+
+		DocumentBuilderFactory docFactory = XmlUtils.getDocumentBuilderFactory();
+		docFactory.setNamespaceAware(true);
+		DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+
+		Document document = docBuilder.parse(new ByteArrayInputStream(docAsXml.getBytes()));
+
+		Document signedAssertion = signAssertion(document, privateKey, certificate);
+
+		String signedAssertionXml = documentToString(signedAssertion);
+
+		return Base64.getEncoder().encodeToString(signedAssertionXml.getBytes());
+	}
+
+	private Document generateSAMLAssertion() throws Exception {
+		String NotBefore = java.time.Instant.now().minusSeconds(60).toString();
+		String NotOnOrAfter = java.time.Instant.now().plusSeconds(session.getSamlAssertionExpiry()).toString();
+		String now = java.time.Instant.now().toString();
+
+		// Create a new XML Document
+		DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+		docFactory.setNamespaceAware(true);
```

Comment:
Klein puntje hoor...
```suggestion
		DocumentBuilderFactory docFactory = XmlUtils.getDocumentBuilderFactory(true);
```